providers/oidc_test.go

```diff
@@ -306,6 +329,13 @@ func TestCreateSessionStateFromBearerToken(t *testing.T) {
 			ExpectedEmail:  defaultIDToken.Email,
 			ExpectedGroups: []string{"test:c", "test:d"},
 		},
+		"Custom Groups Claim2": {
+			IDToken:        customGroupClaimIDToken,
+			GroupsClaim:    "groups",
+			ExpectedUser:   customGroupClaimIDToken.Subject,
+			ExpectedEmail:  customGroupClaimIDToken.Email,
+			ExpectedGroups: []string{"{\"groupId\":\"Admin Group Id\",\"roles\":[\"Admin\"]}"},
```

Comment:
What are your thoughts on getting this structure of a group output to the upstream/downstream in the HTTP Headers?

This PR (https://github.com/oauth2-proxy/oauth2-proxy/pull/799) is working out the challenges of individual headers per group and I think leaning toward comma delimited in a single header to more closely match what other software expects from group RBAC headers.

This might lead to challenges with your IdP's group structure when put into the headers (since it has commas)? How widespread to you think this format is (I didn't realize it was a thing, I hadn't seen it in an OIDC IdP before)?

Reply:
We are using https://www.cidaas.com/de/ as IDP and they are providing the group claim in the structure like above. Regarding #799, I will have a look.

Reply:
I think it is not that widespread, but possible at least with our IDP. At least I did not yet find a standard for the values of the group claim.

Reply:
We have a running joke that every IdP is a special snowflake and breaks conventions/spec in their own special ways 😄 

So I think we just have to do our best to cover 99% of the implementations and handle viable corner cases as they arise.

Reply:
Okay, then let's do it! :)